dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1199 @@
+# -*- coding: utf-8 -*-
+"""Creates a GCP cloud resource tree."""
+
+import json
+import tempfile
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
+from typing import Dict, List, Optional, Any, Set
+from enum import Enum, auto
+
+from dateutil import parser
+from libcloudforensics.providers.gcp.internal.common import CreateService
+from libcloudforensics.providers.gcp.internal.log import GoogleCloudLog
+
+from dftimewolf.lib import module
+from dftimewolf.lib.containers import containers
+from dftimewolf.lib.modules import manager as modules_manager
+from dftimewolf.lib.state import DFTimewolfState
+
+
+class OperatingMode(Enum):
+  """Enum represent operational mode (Online or Offline)."""
+
+  ONLINE = auto()
+  OFFLINE = auto()
+
+
+class Resource:
+  """A Class that represent a resource (Instance, Disk, Image...etc)."""
+
+  has_dynamic_attributes = True  # silences all attribute-errors for Resource
+
+  def __init__(self) -> None:
+    """Initialize the Resource object."""
```

Comment:
please add a comment stating that instant variables of this class represents the fields in a log record and link the docs https://cloud.google.com/compute/docs/logging/migrating-from-activity-logs-to-audit-logs

Reply:
it doesn't necessarily represent the fields in a log record. It could be a disk resource for example obtained using compute api. I already have """A Class that represents a resource (Instance, Disk, Image...etc).""" cause this is a custom class does not necessarily map 1:1 to what's in the proto/log record.